### PR TITLE
fix: use url-parse for url construction

### PR DESCRIPTION
**Pull request by sreya**
This fixes some CodeQL-flagged issues. They're not real issues but the refactor is small and it'll keep the analysis tools quiet.

### CODE REVIEW

**Review by aslilac — CHANGES_REQUESTED** [review]
we should use the built-in `URL` type instead. 

from the url-parse readme:

> url-parse was created in 2014 when the WHATWG URL API was not available in Node.js and the URL interface was supported only in some browsers. Today this is no longer true. The URL interface is available in all supported Node.js release lines and basically all browsers. Consider using it for better security and accuracy.

File: offlinedocs/pages/[[...slug]].tsx [hunk 1]
```diff
@@ -194,8 +194,21 @@ const getNavigation = (manifest: Manifest): Nav => {
...
+
+	// Keep replacing until no more changes occur
+	do {
+		currentResult = lastResult.replace(/<!--[\s\S]*?-->/g, "");
+		if (currentResult === lastResult) {
+			break;
+		}
+		lastResult = currentResult;
+	} while (true);
+	return currentResult;
```
Comment (aslilac) [inline offlinedocs/pages/[[...slug]].tsx:211]:
```suggestion
const removeHtmlComments = (input: string) => {
	return string.replaceAll(/<!--[\s\S]*?-->/g, "");
```

I think this is what you're really after :p

File: site/src/utils/portForward.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import { URL } from "node:url";
```
Comment (aslilac) [inline site/src/utils/portForward.ts]: this code runs in the browser, not node. you can just use the global `URL`, no importing anything.

```suggestion
```

File: offlinedocs/pages/[[...slug]].tsx (hunk 1 above)
Comment (sreya) [reply offlinedocs/pages/[[...slug]].tsx:211]: The problem is that regex doesn't handle nested comments which is what CodeQL is complaining about. The motivation behind the function is to iterate over the string until all comments have been stripped regardless of how nested they are.

Comment (aslilac) [reply offlinedocs/pages/[[...slug]].tsx:211]: the code here, to my eye at least, doesn't do anything to handle "nesting". it just calls `.replace` over and over again until it stops changing the string, which is exactly what `.replaceAll` does

Comment (sreya) [reply offlinedocs/pages/[[...slug]].tsx]: Ahh I missed the `All`. I think that will probably work although I noticed the tool recommended using `html-sanitize` which I think will also solve this and is less likely to trip the tool up in the future.

File: offlinedocs/pages/[[...slug]].tsx [hunk 3]
```diff
@@ -27,7 +27,7 @@ import {
 } from "@chakra-ui/react";
 import fm from "front-matter";
 import { readFileSync } from "fs";
-import _ from "lodash";
+import _, { min } from "lodash";
```
Comment (aslilac) [inline offlinedocs/pages/[[...slug]].tsx:30]:
```suggestion
import _ from "lodash";
```

linter doesn't seem to be calling this out, but this is unused

File: offlinedocs/pages/[[...slug]].tsx [hunk 4]
```diff
@@ -195,7 +196,7 @@ const getNavigation = (manifest: Manifest): Nav => {
 };
 
 const removeHtmlComments = (string: string) => {
-	return string.replace(/<!--[\s\S]*?-->/g, "");
+	return sanitizeHtml(string);
 };
```
Comment (aslilac) [inline offlinedocs/pages/[[...slug]].tsx:200]: at this point, now we're just adding indirection to calling `sanitizeHtml`, and it looks like `removeHtmlComments` is only called in one place. can we just get rid of this function entirely and call `sanitizeHtml` directly there?

**Review by aslilac — APPROVED** [review]
couple small notes, but this looks great 👍

**Pull request merged**